src/components/LocationSearch/LocationSearch.test.jsx

```diff
@@ -0,0 +1,56 @@
+import { describe, it, expect, vi } from "vitest";
+import { render, screen, fireEvent } from "@testing-library/react";
+import LocationSearch from "./LocationSearch";
+
+describe("</ LocationSearch", () => {
+  const renderLocationSearch = (props = {}) => {
+    const defaultProps = {
+      error: "",
+      locations: ["New York", "São Paulo"],
+      isLocationSearchOpen: true,
+      handleLocationSearch: vi.fn(),
+      handleLocationSearchVisibility: vi.fn(),
+    };
+    render(<LocationSearch {...defaultProps} {...props} />);
+  };
+
+  it("renders the error message", () => {
+    renderLocationSearch({ error: "some error here" });
+  });
+
+  it("renders LocationSearch correctly", () => {
+    renderLocationSearch();
+
+    //Checks if the input field is render
+    expect(screen.getByPlaceholderText("search location")).toBeInTheDocument();
```

Comment:
One thing that we usually do in tests as well, is instead of searching by the text with quotes, we use regex, because you can say it's `case insensitive`. In your case, if you happen to change to `Search location` this text would break.

```suggestion
    expect(screen.getByPlaceholderText(/search location/i)).toBeInTheDocument();
```